worker/queueservice.go

```diff
@@ -85,10 +87,11 @@ func (q *queueService) Start() <-chan *taskClaim {
 
 	go func() {
 		for !q.halt.Get() {
-			q.mu.RLock()
-			capacity := q.capacity
-			q.mu.RUnlock()
-			tasks := q.retrieveTasksFromQueue(capacity)
+			claimSize, finished := q.claimSize()
+			if finished {
+				break
```

Comment:
We should break from the loop once we have claimed all of the tasks we are allowed to (i.e. once we hit `maxTasksToRun`).